### PR TITLE
marvell: Continue migration to new APIs

### PR DESCRIPTION
**Pull request by kostapr**
- Fix build issue
- Add initial memory parameters descriptors for BL2
- Migrate to image load V2

Basic build an run test passed on MacchiatoBin board.
Need to fix the service CPU (CM3) image load procesure and test
OPTEE functionality, which probably wiull require additional work.

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>

### CODE REVIEW

**Comment by dp-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by kostapr** [comment]
Oops. Looks like my typo fix interrupted Jenkins. Sorry about it.

File: plat/marvell/common/marvell_bl2_setup.c [hunk 1]
```diff
@@ -193,87 +94,59 @@ void bl2_plat_arch_setup(void)
...
-	bl32_meminfo->free_base = BL32_BASE;
-	bl32_meminfo->total_size =
-			(TRUSTED_DRAM_BASE + TRUSTED_DRAM_SIZE) - BL32_BASE;
-	bl32_meminfo->free_size =
-			(TRUSTED_DRAM_BASE + TRUSTED_DRAM_SIZE) - BL32_BASE;
-}
+	switch (image_id) {
+#ifdef AARCH64
+	case BL32_IMAGE_ID:
+#ifdef SPD_opteed
```
Comment (antonio-nino-diaz-arm) [inline plat/marvell/common/marvell_bl2_setup.c]: Are you actually using OP-TEE?

File: plat/marvell/common/aarch64/marvell_bl2_mem_params_desc.c [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+	    .next_handoff_image_id = BL33_IMAGE_ID,
+    },
+
+	/*
+	 * Fill BL32 external 1 related information.
+	 * A typical use for extra1 image is with OP-TEE where it is the pager image.
+	 */
+    {
+	    .image_id = BL32_EXTRA1_IMAGE_ID,
```
Comment (soby-mathew) [inline plat/marvell/common/aarch64/marvell_bl2_mem_params_desc.c]: As I understand, this is only required for platforms where optee is built with Pageable support. If Marvel has enough memory in Trusted DRAM to load whole of OP-TEE, this support and these additional images may not be required.

File: plat/marvell/common/aarch64/marvell_bl2_mem_params_desc.c [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+	    .next_handoff_image_id = INVALID_IMAGE_ID,
+    },
+
+	/*
+	 * Fill BL32 external 2 related information.
+	 * A typical use for extra2 image is with OP-TEE where it is the paged image.
+	 */
+    {
+	    .image_id = BL32_EXTRA2_IMAGE_ID,
```
Comment (soby-mathew) [inline plat/marvell/common/aarch64/marvell_bl2_mem_params_desc.c]: Same comment as BL32_EXTRA1_IMAGE_ID

File: plat/marvell/common/marvell_bl2_setup.c [hunk 4]
```diff
@@ -193,87 +94,59 @@ void bl2_plat_arch_setup(void)
...
+		pager_mem_params = get_bl_mem_params_node(BL32_EXTRA1_IMAGE_ID);
+		assert(pager_mem_params);
+
+		paged_mem_params = get_bl_mem_params_node(BL32_EXTRA2_IMAGE_ID);
```
Comment (soby-mathew) [inline plat/marvell/common/marvell_bl2_setup.c]: From the previous diff, it looks like you don't need this extra Optee images.

File: plat/marvell/common/marvell_bl2_setup.c (hunk 1 above)
Comment (kostapr) [reply plat/marvell/common/marvell_bl2_setup.c]: Yes, we support it. At least supported in v1.5 :-)

File: plat/marvell/common/aarch64/marvell_bl2_mem_params_desc.c (hunk 2 above)
Comment (kostapr) [reply plat/marvell/common/aarch64/marvell_bl2_mem_params_desc.c]: I see. Our OP-TEE maintainer left the company, so now I should take care of this code, which I am currently not familiar with. For this patch I just took the ARM sources as a base. But I am sure we do have enough memory for OP-TEE load.

File: plat/marvell/common/marvell_bl2_setup.c (hunk 4 above)
Comment (kostapr) [reply plat/marvell/common/marvell_bl2_setup.c]: OK, I will remove these descriptors

**Comment by soby-mathew** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
The tests related to your platforms have passed.

**Pull request merged**